src/test/java/org/sqlite/DBMetaDataTest.java

```diff
@@ -1669,4 +1667,593 @@ public void getPrimaryKeys(String table) throws SQLException {
+        }
+
+        @Test
+        public void getTablesForAttachedDatabase() throws SQLException {
+            ResultSet rs = meta.getTables(null, "db2", null, null);
+            assertThat(rs).isNotNull();
+
+            stat.getGeneratedKeys().close();
+            stat.close();
+
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_NAME")).isEqualTo("sqlite_schema");
+            assertThat(rs.getString("TABLE_TYPE")).isEqualTo("SYSTEM TABLE");
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_NAME")).isEqualTo("test2"); // 3
+            assertThat(rs.getString("TABLE_TYPE")).isEqualTo("TABLE"); // 4
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_NAME")).isEqualTo("testView2");
+            assertThat(rs.getString("TABLE_TYPE")).isEqualTo("VIEW");
+            assertThat(rs.next()).isFalse();
+            rs.close();
+
+            rs = meta.getTables(null, null, "bob", null);
+            assertThat(rs.next()).isFalse();
+            rs.close();
+            rs = meta.getTables(null, null, "test", null);
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.next()).isFalse();
+            rs.close();
+            rs = meta.getTables(null, null, "test%", null);
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.next()).isTrue();
+            rs.close();
+
+            rs = meta.getTables(null, "main", null, new String[] {"table"});
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_NAME")).isEqualTo("test");
+            assertThat(rs.next()).isFalse();
+            rs.close();
+
+            rs = meta.getTables(null, "main", null, new String[] {"view"});
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_NAME")).isEqualTo("testView");
+            assertThat(rs.next()).isFalse();
+            rs.close();
+
+            rs = meta.getTables(null, "main", null, new String[] {"system table"});
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_NAME")).isEqualTo("sqlite_schema");
+            assertThat(rs.next()).isFalse();
+            rs.close();
+        }
+
+        @Test
+        public void testDynamicDatabaseAttachment() throws IOException, SQLException {
+            ResultSet schemas;
+            File thirdDB = Files.createTempFile("db3", ".sqlite").toFile();
+            stat.executeUpdate("attach database \"" + thirdDB.toURI().toURL() + "\" as db3;");
+            try {
+                schemas = meta.getSchemas();
+                boolean schemaFound = false;
+                while (schemas.next()) {
+                    schemaFound = "db3".equals(schemas.getString("TABLE_SCHEM"));
+                    if (schemaFound) {
+                        break;
+                    }
+                }
+                assertThat(schemaFound).isTrue();
```

Comment:
maybe put all schemas in a list and verify that it contains all the values we expect, ie `db2`,`db3` and `main`.
If we expect some ordering, we should also test for that here.